addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/ChannelsTypeProviderAvailableInputs.java

```diff
@@ -72,35 +68,63 @@ public ChannelTypeUID getChannelTypeUID() {
     public ChannelsTypeProviderAvailableInputs(ThingUID thing) {
         channelTypeUID = new ChannelTypeUID(YamahaReceiverBindingConstants.BINDING_ID,
                 YamahaReceiverBindingConstants.CHANNEL_INPUT_TYPE_AVAILABLE + thing.getId());
+        createChannelType(getDefaultStateDescription());
+    }
 
-        // Use the INPUT channel type as a template to create a new channel type AVAILABLE_INPUTS.
-        final ChannelTypeUID channelTypeInputsUID = new ChannelTypeUID(YamahaReceiverBindingConstants.BINDING_ID,
-                YamahaReceiverBindingConstants.CHANNEL_INPUT_TYPE_DEFAULT);
-        templateType = TypeResolver.resolve(channelTypeInputsUID);
-        if (templateType == null) {
-            logger.warn("Couldn't find: {}", channelTypeInputsUID);
-        } else {
-            channelType = new ChannelType(channelTypeUID, false, "String", templateType.getKind(),
-                    templateType.getLabel(), templateType.getDescription(), templateType.getCategory(),
-                    templateType.getTags(), templateType.getState(), templateType.getEvent(),
-                    templateType.getConfigDescriptionURI());
-        }
+    private void createChannelType(StateDescription state) {
+        channelType = new ChannelType(channelTypeUID, false, "String", ChannelKind.STATE, "Input source",
```

Comment:
Are those strings (and the StateOption("...", "Net Radio") ones) still translatable? I like the xml-less solution.

Reply:
@sjka wdyt?

> Are those strings (and the StateOption("...", "Net Radio") ones) still translatable? I like the xml-less solution.

Reply:
@sjka can you respond to @davidgraeff his question.